doc(pytorch): suggest using external_modules to save models (#3097)
bentoml/BentoML: docs/source/frameworks/pytorch.rst

.. note::

>     :bdg-info:`Remarks:` All of the external python classes/util functions that the model
>     requires must be referenced in `<module>.<class>` format, and ensure the module is passed
>     to ``bentoml.pytorch.save`` as the ``external_modules`` parameter. For example:

Reviewer: ```suggestion
    :bdg-info:`Remarks:` External python classes or utility functions required by the model must be referenced in ``<module>.<class>`` format, and such modules should be passed to ``bentoml.pytorch.save`` via ``external_modules``. For example:
```

>     .. code-block:: python
>         :caption: `train.py`
>         
>         import my_models  # where your model class is defined
> 
>         model = my_models.MyModel()
>         bentoml.pytorch.save(
>             model,
>             "my_torch_model",
>             signatures={"__call__": {"batchable": True, "batchdim": 0}},
>             external_modules=[my_models],
>         )

Reviewer: ```suggestion
    .. code-block:: python
       :caption: `train.py`
       :emphasize-lines: 1,8
        
       import my_models

       model = my_models.MyModel()
       bentoml.pytorch.save(
           model,
           "my_torch_model",
           signatures={"__call__": {"batchable": True, "batch_dim": 0}},
           external_modules=[my_models],
       )
```

>     The PyTorch framework requires them to restore the model. This is due to the limitation of PyTorch model serialization.
>     A better practice is to use :doc:`/frameworks/torchscript` to convert the model to TorchScript, which is more portable and less prone to compatibility issues.

Reviewer: ```suggestion
    This is due to a limitation from PyTorch for model serialisation, where PyTorch needs to use the model's source code to be able to restore it.
    
    A better practice is to use :doc:`/frameworks/torchscript`, which will be compiled to `TorchScript <https://pytorch.org/docs/stable/jit.html>`_ format.
```
